### PR TITLE
Specific contract type used

### PR DESCRIPTION
**Pull request by satyamakgec**

### CODE REVIEW

File: contracts/libraries/TokenLib.sol [hunk 1]
```diff
@@ -1,13 +1,13 @@
 pragma solidity ^0.5.0;
 
+import "./KindMath.sol";
```
Comment (maxsam4) [inline contracts/libraries/TokenLib.sol]: This may lead to adding back `KindMath` import after the merge.

Comment (satyamakgec) [reply contracts/libraries/TokenLib.sol]: I think this will be solved by the other PR.

Comment (maxsam4) [reply contracts/libraries/TokenLib.sol]: That pr removes it but this pr ads it back. (it's in green not white). This pr is specifically adding it back.

File: contracts/libraries/TokenLib.sol [hunk 2]
```diff
@@ -231,19 +230,19 @@ library TokenLib {
         address _module,
         uint256 _change,
         bool _increase,
-        address _polyToken,
+        IERC20 _polyToken,
```
Comment (tintinweb) [inline contracts/libraries/TokenLib.sol:236]: Note: could be `IPoly` to avoid casting on line 242,245 but I see that this is for consistency as polyToken is usually used as IERC20 without ever calling increase/decreaseApproval (which is only part of `IPoly`). Is probably cleaner the way it is right now keeping polyToken an IERC20 and only casting it to IPoly for increaseApproval :)

File: contracts/tokens/SecurityToken.sol [hunk 3]
```diff
@@ -1061,9 +1057,9 @@ contract SecurityToken is ERC20, ReentrancyGuard, SecurityTokenStorage, IERC1594
 
     function updateFromRegistry() public {
         _onlyOwner();
-        moduleRegistry = PolymathRegistry(polymathRegistry).getAddress("ModuleRegistry");
-        securityTokenRegistry = PolymathRegistry(polymathRegistry).getAddress("SecurityTokenRegistry");
-        polyToken = PolymathRegistry(polymathRegistry).getAddress("PolyToken");
+        moduleRegistry = IModuleRegistry(IPolymathRegistry(polymathRegistry).getAddress("ModuleRegistry"));
```
Comment (tintinweb) [inline contracts/tokens/SecurityToken.sol]: might be unnecessary to cast the registry to `IPolymathRegistry` as it is already declared as `IPolymathRegistry` in `SecurityTokenStorage`.

same in `UpgradeableModuleFactory::76` (not part of this PR) where `polymathRegistry` is a statevar inherited from `ModuleFactory` declared as `IPolymathRegistry` already

**Comment by tintinweb** [comment]
one minor note for what appears to be a double cast - please verify. most other issues (eg. address comparison should be caught by the compiler). good to merge otherwise.

tin/diligence

**Comment by tintinweb** [comment]
👍 tin/diligence

**Review by maxsam4 — APPROVED** [review]

**Comment by maxsam4** [comment]
@satyamakgec Please fix the merge conflicts.
